urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Configuration.java

```diff
@@ -0,0 +1,335 @@
+    private static Properties defaultProperty;
+
+    static {
+        init();
+    }
+
+    /*package*/ static void init() {
+        defaultProperty = new Properties();
+        defaultProperty.setProperty("weibo4j.debug", "true");
+        defaultProperty.setProperty("weibo4j.source", Weibo.CONSUMER_KEY);
+        //defaultProperty.setProperty("weibo4j.clientVersion","");
+        defaultProperty.setProperty("weibo4j.clientURL", "http://open.t.sina.com.cn/-{weibo4j.clientVersion}.xml");
+        defaultProperty.setProperty("weibo4j.http.userAgent", "weibo4j http://open.t.sina.com.cn/ /{weibo4j.clientVersion}");
+        //defaultProperty.setProperty("weibo4j.user","");
+        //defaultProperty.setProperty("weibo4j.password","");
+        defaultProperty.setProperty("weibo4j.http.useSSL", "false");
+        //defaultProperty.setProperty("weibo4j.http.proxyHost","");
+        defaultProperty.setProperty("weibo4j.http.proxyHost.fallback", "http.proxyHost");
+        //defaultProperty.setProperty("weibo4j.http.proxyUser","");
+        //defaultProperty.setProperty("weibo4j.http.proxyPassword","");
+        //defaultProperty.setProperty("weibo4j.http.proxyPort","");
+        defaultProperty.setProperty("weibo4j.http.proxyPort.fallback", "http.proxyPort");
+        defaultProperty.setProperty("weibo4j.http.connectionTimeout", "20000");
+        defaultProperty.setProperty("weibo4j.http.readTimeout", "120000");
+        defaultProperty.setProperty("weibo4j.http.retryCount", "3");
+        defaultProperty.setProperty("weibo4j.http.retryIntervalSecs", "10");
+        //defaultProperty.setProperty("weibo4j.oauth.consumerKey","");
+        //defaultProperty.setProperty("weibo4j.oauth.consumerSecret","");
+        defaultProperty.setProperty("weibo4j.async.numThreads", "1");
+        defaultProperty.setProperty("weibo4j.clientVersion", Version.getVersion());
+        try {
+            // Android platform should have dalvik.system.VMRuntime in the classpath.
+            // @see http://developer.android.com/reference/dalvik/system/VMRuntime.html
+            Class.forName("dalvik.system.VMRuntime");
+            defaultProperty.setProperty("weibo4j.dalvik", "true");
+        } catch (ClassNotFoundException cnfe) {
+            defaultProperty.setProperty("weibo4j.dalvik", "false");
+        }
+        DALVIK = getBoolean("weibo4j.dalvik");
+        String t4jProps = "weibo4j.properties";
+        boolean loaded = loadProperties(defaultProperty, "." + File.separatorChar + t4jProps) ||
+                loadProperties(defaultProperty, Configuration.class.getResourceAsStream("/WEB-INF/" + t4jProps)) ||
+                loadProperties(defaultProperty, Configuration.class.getResourceAsStream("/" + t4jProps));
+    }
+
+    private static boolean loadProperties(Properties props, String path) {
+        try {
+            File file = new File(path);
+            if(file.exists() && file.isFile()){
+                props.load(new FileInputStream(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).